diff --git a/internal/Util.hpp b/internal/Util.hpp
index 5f99f87..b7b9545 100644
--- a/internal/Util.hpp
+++ b/internal/Util.hpp
@@ -178,7 +178,7 @@ static bool loadMoFileStringsToArray(FILE* moFile,
 			{
 				firstStringOffset = currentStringOffset;
 			}
-			
+
 			if (i == (numberOfStrings - 1))
 			{
 				lastStringOffset = currentStringOffset;
@@ -216,10 +216,10 @@ static bool loadMoFileStringsToArray(FILE* moFile,
 	return true;
 }
 
-static std::vector<std::string> buildMoFilePaths(const char *domain)
+static std::vector<std::string> buildMoFilePaths(const char *domain, std::string& userLanguage)
 {
 	const char *pLanguage = getenv("LANGUAGE");
-	std::string languages = pLanguage ? pLanguage : "";
+	std::string languages = userLanguage.empty() ? (pLanguage ? pLanguage : "") : userLanguage;
 	std::vector<std::string> paths {""};
 	std::string slash = "/";
 	const size_t length = languages.size();
diff --git a/internal/libintl.cpp b/internal/libintl.cpp
index 186c16a..ba612f0 100644
--- a/internal/libintl.cpp
+++ b/internal/libintl.cpp
@@ -48,249 +48,212 @@ using namespace libintllite::internal;
 
 static char* currentDefaultDomain = NULL;
 static map<char*, MessageCatalog*> loadedMessageCatalogPtrsByDomain;
+static std::string g_language{};
 
-libintl_lite_bool_t loadMessageCatalog(const char* domain, const char* moFilesRoot)
-{
-	if (!moFilesRoot || !domain)
-	{
-		return LIBINTL_LITE_BOOL_FALSE;
-	}
-
-	FILE* moFile = NULL;
-	CloseFileHandleGuard closeFileHandleGuard(moFile);
-
-	for (const string &path : buildMoFilePaths(domain))
-	{
-		string fullPath = moFilesRoot + path;
-		moFile = fopen(fullPath.c_str(), "rb");
-		if (loadMessageCatalogFile(domain, moFile) == LIBINTL_LITE_BOOL_TRUE)
-		{
-			return LIBINTL_LITE_BOOL_TRUE;
-		}
-	}
-
-	return LIBINTL_LITE_BOOL_FALSE;
+libintl_lite_bool_t loadMessageCatalog(const char* domain, const char* moFilesRoot, std::string& userLanguage) {
+    if (!moFilesRoot || !domain) {
+        return LIBINTL_LITE_BOOL_FALSE;
+    }
+
+    FILE* moFile = NULL;
+    CloseFileHandleGuard closeFileHandleGuard(moFile);
+
+    for (const string& path: buildMoFilePaths(domain, userLanguage)) {
+        string fullPath = moFilesRoot + path;
+        moFile = fopen(fullPath.c_str(), "rb");
+        if (loadMessageCatalogFile(domain, moFile) == LIBINTL_LITE_BOOL_TRUE) {
+            return LIBINTL_LITE_BOOL_TRUE;
+        }
+    }
+
+    return LIBINTL_LITE_BOOL_FALSE;
+}
+
+libintl_lite_bool_t loadMessageCatalogFile(const char* domain, FILE* moFile) {
+    try {
+        if (sizeof(uint32_t) != 4) {
+            return LIBINTL_LITE_BOOL_FALSE;
+        }
+
+        if (!moFile || !domain) {
+            return LIBINTL_LITE_BOOL_FALSE;
+        }
+
+        uint32_t magicNumber;
+        if (!readUIn32FromFile(moFile, false, magicNumber)) return LIBINTL_LITE_BOOL_FALSE;
+        if ((magicNumber != 0x950412de) && (magicNumber != 0xde120495)) return LIBINTL_LITE_BOOL_FALSE;
+
+        uint32_t fileFormatRevision;
+        if (!readUIn32FromFile(moFile, false, fileFormatRevision)) return LIBINTL_LITE_BOOL_FALSE;
+        if (fileFormatRevision != 0) return LIBINTL_LITE_BOOL_FALSE;
+
+        bool needsBeToLeConversion = isBigEndian();
+
+        uint32_t numberOfStrings;
+        if (!readUIn32FromFile(moFile, needsBeToLeConversion, numberOfStrings)) return false;
+        if (numberOfStrings == 0) {
+            return LIBINTL_LITE_BOOL_TRUE;
+        }
+
+        uint32_t offsetOrigTable;
+        if (!readUIn32FromFile(moFile, needsBeToLeConversion, offsetOrigTable)) return LIBINTL_LITE_BOOL_FALSE;
+
+        uint32_t offsetTransTable;
+        if (!readUIn32FromFile(moFile, needsBeToLeConversion, offsetTransTable)) return LIBINTL_LITE_BOOL_FALSE;
+
+        string* sortedOrigStringsArray = NULL;
+        ArrayGurard<string> sortedOrigStringsArrayGuard(sortedOrigStringsArray);
+        sortedOrigStringsArray = new string[numberOfStrings];
+        if (!sortedOrigStringsArray) {
+            return LIBINTL_LITE_BOOL_FALSE;
+        }
+
+        if (!loadMoFileStringsToArray(moFile,
+                                      numberOfStrings,
+                                      offsetOrigTable,
+                                      needsBeToLeConversion,
+                                      sortedOrigStringsArray)) return LIBINTL_LITE_BOOL_FALSE;
+
+        string* translatedStringsArray = NULL;
+        ArrayGurard<string> translatedStringsArrayGuard(translatedStringsArray);
+        translatedStringsArray = new string[numberOfStrings];
+        if (!translatedStringsArray) {
+            return LIBINTL_LITE_BOOL_FALSE;
+        }
+
+        if (!loadMoFileStringsToArray(moFile,
+                                      numberOfStrings,
+                                      offsetTransTable,
+                                      needsBeToLeConversion,
+                                      translatedStringsArray)) return LIBINTL_LITE_BOOL_FALSE;
+
+        MessageCatalog* newMessageCatalogPtr = new MessageCatalog(numberOfStrings,
+                                                                  sortedOrigStringsArray,
+                                                                  translatedStringsArray);
+        if (!newMessageCatalogPtr) return LIBINTL_LITE_BOOL_FALSE;
+        sortedOrigStringsArrayGuard.release();
+        translatedStringsArrayGuard.release();
+
+        char* domainDup = strdup(domain);
+        if (!domainDup) return LIBINTL_LITE_BOOL_FALSE;
+        closeLoadedMessageCatalog(domain);
+        loadedMessageCatalogPtrsByDomain[domainDup] = newMessageCatalogPtr;
+
+        return LIBINTL_LITE_BOOL_TRUE;
+    } catch (...) {
+        return LIBINTL_LITE_BOOL_FALSE;
+    }
+}
+
+libintl_lite_bool_t bindtextdomain(const char* domain, const char* moFilePath) {
+    return loadMessageCatalog(domain, moFilePath, g_language);
 }
 
-libintl_lite_bool_t loadMessageCatalogFile(const char* domain, FILE* moFile)
-{
-	try
-	{
-		if (sizeof(uint32_t) != 4)
-		{
-			return LIBINTL_LITE_BOOL_FALSE;
-		}
-
-		if (!moFile || !domain)
-		{
-			return LIBINTL_LITE_BOOL_FALSE;
-		}
-
-		uint32_t magicNumber;
-		if (!readUIn32FromFile(moFile, false, magicNumber)) return LIBINTL_LITE_BOOL_FALSE;
-		if ((magicNumber != 0x950412de) && (magicNumber != 0xde120495)) return LIBINTL_LITE_BOOL_FALSE;
-
-		uint32_t fileFormatRevision;
-		if (!readUIn32FromFile(moFile, false, fileFormatRevision)) return LIBINTL_LITE_BOOL_FALSE;
-		if (fileFormatRevision != 0) return LIBINTL_LITE_BOOL_FALSE;
-
-		bool needsBeToLeConversion = isBigEndian();
-
-		uint32_t numberOfStrings;
-		if (!readUIn32FromFile(moFile, needsBeToLeConversion, numberOfStrings)) return false;
-		if (numberOfStrings == 0)
-		{
-			return LIBINTL_LITE_BOOL_TRUE;
-		}
-
-		uint32_t offsetOrigTable;
-		if (!readUIn32FromFile(moFile, needsBeToLeConversion, offsetOrigTable)) return LIBINTL_LITE_BOOL_FALSE;
-
-		uint32_t offsetTransTable;
-		if (!readUIn32FromFile(moFile, needsBeToLeConversion, offsetTransTable)) return LIBINTL_LITE_BOOL_FALSE;
-
-		string* sortedOrigStringsArray = NULL;
-		ArrayGurard<string> sortedOrigStringsArrayGuard(sortedOrigStringsArray);
-		sortedOrigStringsArray = new string[numberOfStrings];
-		if (!sortedOrigStringsArray)
-		{
-			return LIBINTL_LITE_BOOL_FALSE;
-		}
-
-		if (!loadMoFileStringsToArray(moFile,
-				numberOfStrings,
-				offsetOrigTable,
-				needsBeToLeConversion,
-				sortedOrigStringsArray)) return LIBINTL_LITE_BOOL_FALSE;
-
-		string* translatedStringsArray = NULL;
-		ArrayGurard<string> translatedStringsArrayGuard(translatedStringsArray);
-		translatedStringsArray = new string[numberOfStrings];
-		if (!translatedStringsArray)
-		{
-			return LIBINTL_LITE_BOOL_FALSE;
-		}
-
-		if (!loadMoFileStringsToArray(moFile,
-				numberOfStrings,
-				offsetTransTable,
-				needsBeToLeConversion,
-				translatedStringsArray)) return LIBINTL_LITE_BOOL_FALSE;
-
-		MessageCatalog* newMessageCatalogPtr = new MessageCatalog(numberOfStrings,
-				sortedOrigStringsArray,
-				translatedStringsArray);
-		if (!newMessageCatalogPtr) return LIBINTL_LITE_BOOL_FALSE;
-		sortedOrigStringsArrayGuard.release();
-		translatedStringsArrayGuard.release();
-
-		char* domainDup = strdup(domain);
-		if (!domainDup) return LIBINTL_LITE_BOOL_FALSE;
-		closeLoadedMessageCatalog(domain);
-		loadedMessageCatalogPtrsByDomain[domainDup] = newMessageCatalogPtr;
-
-		return LIBINTL_LITE_BOOL_TRUE;
-	}
-	catch (...)
-	{
-		return LIBINTL_LITE_BOOL_FALSE;
-	}
+libintl_lite_bool_t bind_textdomain_codeset(const char* domain, const char* moFilePath) {
+    // not implemented yet
+    return LIBINTL_LITE_BOOL_FALSE;
 }
 
-libintl_lite_bool_t bindtextdomain(const char* domain, const char* moFilePath)
-{
-	return loadMessageCatalog( domain, moFilePath );
+void closeLoadedMessageCatalog(const char* domain) {
+    if (domain) {
+        for (map<char*, MessageCatalog*>::iterator i = loadedMessageCatalogPtrsByDomain.begin();
+             i != loadedMessageCatalogPtrsByDomain.end();
+             i++) {
+            if (strcmp(i->first, domain) == 0) {
+                free(i->first);
+                delete i->second;
+                loadedMessageCatalogPtrsByDomain.erase(i);
+                return;
+            }
+        }
+    }
 }
 
-libintl_lite_bool_t bind_textdomain_codeset(const char* domain, const char* moFilePath)
-{
-	// not implemented yet
-	return LIBINTL_LITE_BOOL_FALSE;
+void closeAllLoadedMessageCatalogs() {
+    for (map<char*, MessageCatalog*>::iterator i = loadedMessageCatalogPtrsByDomain.begin();
+         i != loadedMessageCatalogPtrsByDomain.end();
+         i++) {
+        free(i->first);
+        delete i->second;
+    }
+    loadedMessageCatalogPtrsByDomain.clear();
+    free(currentDefaultDomain);
+    currentDefaultDomain = NULL;
 }
 
-void closeLoadedMessageCatalog(const char* domain)
-{
-	if (domain)
-	{
-		for (map<char*, MessageCatalog*>::iterator i = loadedMessageCatalogPtrsByDomain.begin();
-				i != loadedMessageCatalogPtrsByDomain.end();
-				i++)
-		{
-			if (strcmp(i->first, domain) == 0)
-			{
-				free(i->first);
-				delete i->second;
-				loadedMessageCatalogPtrsByDomain.erase(i);
-				return;
-			}
-		}
-	}
+const char* textdomain(const char* domain) {
+    if (domain) {
+        char* newDefaultDomain = strdup(domain);
+        if (!newDefaultDomain) {
+            return NULL;
+        }
+        free(currentDefaultDomain);
+        currentDefaultDomain = newDefaultDomain;
+        return newDefaultDomain;
+    } else {
+        return NULL;
+    }
 }
 
-void closeAllLoadedMessageCatalogs()
-{
-	for (map<char*, MessageCatalog*>::iterator i = loadedMessageCatalogPtrsByDomain.begin();
-			i != loadedMessageCatalogPtrsByDomain.end();
-			i++)
-	{
-		free(i->first);
-		delete i->second;
-	}
-	loadedMessageCatalogPtrsByDomain.clear();
-	free(currentDefaultDomain);
-	currentDefaultDomain = NULL;
+const char* gettext(const char* origStr) {
+    return dgettext(NULL, origStr);
 }
 
-const char* textdomain(const char* domain)
-{
-	if (domain)
-	{
-		char* newDefaultDomain = strdup(domain);
-		if (!newDefaultDomain)
-		{
-			return NULL;
-		}
-		free(currentDefaultDomain);
-		currentDefaultDomain = newDefaultDomain;
-		return newDefaultDomain;
-	}
-	else
-	{
-		return NULL;
-	}
+const char* dgettext(const char* domain, const char* origStr) {
+    if (!origStr) {
+        return NULL;
+    }
+
+    if (!domain) {
+        if (currentDefaultDomain) {
+            domain = currentDefaultDomain;
+        } else {
+            return origStr;
+        }
+    }
+
+    const MessageCatalog* msgCat = NULL;
+    for (map<char*, MessageCatalog*>::iterator i = loadedMessageCatalogPtrsByDomain.begin();
+         !msgCat && (i != loadedMessageCatalogPtrsByDomain.end());
+         i++) {
+        if (strcmp(i->first, domain) == 0) {
+            msgCat = i->second;
+        }
+    }
+
+    if (!msgCat) {
+        return origStr;
+    }
+
+    const string* translatedStrPtr = msgCat->getTranslatedStrPtr(origStr);
+    if (translatedStrPtr) {
+        return translatedStrPtr->c_str();
+    } else {
+        return origStr;
+    }
 }
 
-const char* gettext(const char* origStr)
-{
-	return dgettext(NULL, origStr);
+const char* ngettext(const char* origStr, const char* origStrPlural, unsigned long n) {
+    if (n == 1) {
+        return gettext(origStr);
+    } else {
+        return gettext(origStrPlural);
+    }
 }
 
-const char* dgettext(const char* domain, const char* origStr)
-{
-	if (!origStr)
-	{
-		return NULL;
-	}
-
-	if (!domain)
-	{
-		if (currentDefaultDomain)
-		{
-			domain = currentDefaultDomain;
-		}
-		else
-		{
-			return origStr;
-		}
-	}
-
-	const MessageCatalog* msgCat = NULL;
-	for (map<char*, MessageCatalog*>::iterator i = loadedMessageCatalogPtrsByDomain.begin();
-			!msgCat && (i != loadedMessageCatalogPtrsByDomain.end());
-			i++)
-	{
-		if (strcmp(i->first, domain) == 0)
-		{
-			msgCat = i->second;
-		}
-	}
-
-	if (!msgCat)
-	{
-		return origStr;
-	}
-
-	const string* translatedStrPtr = msgCat->getTranslatedStrPtr(origStr);
-	if (translatedStrPtr)
-	{
-		return translatedStrPtr->c_str();
-	}
-	else
-	{
-		return origStr;
-	}
+const char* dngettext(const char* domain, const char* origStr, const char* origStrPlural, unsigned long n) {
+    if (n == 1) {
+        return dgettext(domain, origStr);
+    } else {
+        return dgettext(domain, origStrPlural);
+    }
 }
 
-const char* ngettext(const char* origStr, const char* origStrPlural, unsigned long n)
-{
-	if (n == 1)
-	{
-		return gettext(origStr);
-	}
-	else
-	{
-		return gettext(origStrPlural);
-	}
+void setlanguage(const char* language) {
+    g_language = language;
 }
 
-const char* dngettext(const char* domain, const char* origStr, const char* origStrPlural, unsigned long n)
-{
-	if (n == 1)
-	{
-		return dgettext(domain, origStr);
-	}
-	else
-	{
-		return dgettext(domain, origStrPlural);
-	}
+
+std::string getlanguage() {
+    return g_language;
 }
diff --git a/libintl.h b/libintl.h
index f022415..c7e08ea 100644
--- a/libintl.h
+++ b/libintl.h
@@ -117,6 +117,8 @@ LIBINTL_LITE_API const char* ngettext(const char* origStr, const char* origStrPl
  */
 LIBINTL_LITE_API const char* dngettext(const char* domain, const char* origStr, const char* origStrPlural, unsigned long n);
 
+LIBINTL_LITE_API void setlanguage(const char* language);
+
 #ifdef __cplusplus
 } // extern "C"
 #endif
